components/vocabularies/api/src/main/java/org/phenotips/vocabulary/internal/DefaultVocabularyManager.java

```diff
@@ -86,6 +146,83 @@ public Vocabulary getVocabulary(String vocabularyId)
         return new ArrayList<>(this.vocabularies.keySet());
     }
 
+    @Override
+    public List<VocabularyTerm> search(@Nullable final String input, @Nullable final String category,
+        final int maxResults)
+    {
+        // If either the category or the input is blank, then there is nothing to search. Return empty list.
+        if (StringUtils.isBlank(input) || StringUtils.isBlank(category)) {
+            return Collections.emptyList();
+        }
+        // Try to get categorizedVocabularies that belong to the provided category. If none returned, return empty list.
+        final Set<Vocabulary> categorizedVocabularies = this.vocabulariesByCategory.get(category);
+        if (CollectionUtils.isEmpty(categorizedVocabularies)) {
+            this.logger.warn("No vocabularies associated with the specified category: {}", category);
+            return Collections.emptyList();
+        }
+
+        return search(input, maxResults, category, categorizedVocabularies);
+    }
+
+    /**
+     * Performs a search for {@code input query string} using the provided set of {@code categorizedVocabularies}, and
+     * returns the specified {@code maxResults number of results}, sorted by score (in descending order).
+     *
+     * @param input the input query string
+     * @param maxResults the maximum number of results to return
+     * @param category the vocabulary category
+     * @param categorizedVocabularies the {@link Vocabulary} objects to search {@code input} in
+     * @return an list of {@link VocabularyTerm} objects to search in
+     */
+    private List<VocabularyTerm> search(@Nonnull final String input, final int maxResults,
+        @Nonnull final String category, @Nonnull final Set<Vocabulary> categorizedVocabularies)
+    {
+        final List<VocabularyTerm> results = new ArrayList<>();
+        for (final Vocabulary vocabulary : categorizedVocabularies) {
+            results.addAll(vocabulary.search(input, category, maxResults, SORT_ORDER, null));
```

Comment:
I'm not sure specifying that results should be sorted by score is needed, that's the default.